src/sql-parser/src/parser.rs

```diff
@@ -1160,6 +1167,92 @@ impl<'a> Parser<'a> {
         }))
     }
 
+    fn parse_operator_expr(&mut self) -> Result<Token, ParserError> {
+        self.expect_token(&Token::LParen)?;
+        let tok = self
+            .next_token()
+            .ok_or_else(|| self.error(self.peek_prev_pos(), "Unexpected EOF".to_string()))?;
+
+        let operation = match tok {
+            Token::Op(..) => Ok(tok),
+            Token::Star => Ok(tok),
+            Token::Ident(schema) => {
+                if schema.ne("pg_catalog") {
+                    return self.expected(self.peek_prev_pos(), "unknown schema", Some(Token::Dot));
+                }
+                self.expect_token(&Token::Dot)?;
+
+                let inner_tok = self.next_token().ok_or_else(|| {
+                    self.error(self.peek_prev_pos(), "Unexpected EOF".to_string())
+                })?;
+
+                match inner_tok {
+                    Token::Op(..) => Ok(inner_tok),
```

Comment:
This has the side effect of removing the `OPERATOR(x)` from the parse tree, which does wrong things for precedence. You need to add a new Op variant to handle this case, like the suggested `Op::PgCatalogQualified(Box<Op>)`.